src/wallet/rpcwallet.cpp

```diff
@@ -2534,13 +2535,26 @@ UniValue z_listunspent(const UniValue& params, bool fHelp)
             string address = o.get_str();
             auto zaddr = DecodePaymentAddress(address);
             if (IsValidPaymentAddress(zaddr)) {
-                // TODO: Add Sapling support. For now, ensure we can freely convert.
-                assert(boost::get<libzcash::SproutPaymentAddress>(&zaddr) != nullptr);
-                libzcash::SproutPaymentAddress addr = boost::get<libzcash::SproutPaymentAddress>(zaddr);
-                if (!fIncludeWatchonly && !pwalletMain->HaveSproutSpendingKey(addr)) {
+                // TODO: Add visitor to handle support for Sprout and Sapling addrs.
```

Comment:
In this particular case, this has already been done see `class HaveSpendingKeyForPaymentAddress : public boost::static_visitor<bool>` `in wallet.h` It would be good to reuse that below as this could replace this whole if .. else if .. else block.